Framework/Algorithms/inc/MantidAlgorithms/XrayAbsorptionCorrection.h

```diff
@@ -0,0 +1,72 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+//------------------------------------------------------------------------------
+// Includes
+//------------------------------------------------------------------------------
+#include "MantidAPI/Algorithm.h"
+#include "MantidAlgorithms/InterpolationOption.h"
+#include "MantidAlgorithms/SampleCorrections/IBeamProfile.h"
+#include "MantidAlgorithms/SampleCorrections/MCAbsorptionStrategy.h"
+#include "MantidAlgorithms/SampleCorrections/MCInteractionVolume.h"
+#include "MantidAlgorithms/SampleCorrections/SparseWorkspace.h"
+
+namespace Mantid {
+namespace API {
+class Sample;
+}
+namespace Geometry {
+class Instrument;
+}
+
+namespace Algorithms {
+class DetectorGridDefinition;
+class MCInteractionVolume;
+
+/**
+ *Calculates attenuation of xrays due to absorption in a sample.
+ */
+class MANTID_ALGORITHMS_DLL XrayAbsorptionCorrection : public API::Algorithm {
+public:
+  /// Algorithm's name
+  const std::string name() const override { return "XrayAbsorptionCorrection"; }
+  /// Algorithm's version
+  int version() const override { return 1; }
+  /// Algorithm's category for identification
+  const std::string category() const override {
+    return "CorrectionFunctions\\AbsorptionCorrections";
+  }
+  /// Summary of algorithms purpose
+  const std::string summary() const override {
+    return "Calculates attenuation of xrays produced by negative muons due"
+           " to absorption in a sample for elemental analysis ";
+  }
+  const std::vector<std::string> seeAlso() const override {
+    return {"ConvertToPointData"};
```

Comment:
I think you've added this in response to Anthony's request? This is normally for similar algorithms or other algorithms the user might run together with this one rather than algorithms called inside the .cpp unit. Perhaps the SetSample algorithm might be more useful because that's where the user sets up the X-ray attenuation profiles?